internal/pkg/plugin/plugin.go

```diff
@@ -183,15 +184,30 @@ func (p *AMDGPUPlugin) ListAndWatch(e *pluginapi.Empty, s pluginapi.DevicePlugin
 		select {
 		case <-p.Heartbeat:
 			var health = pluginapi.Unhealthy
+			var hasHealthSvc = false
 
-			// TODO there are no per device health check currently
-			// TODO all devices on a node is used together by kfd
 			if simpleHealthCheck() {
 				health = pluginapi.Healthy
 			}
 
+			// per device health check
```

Comment:
Can this whole advanced health check be moved inside another function, which gets called from here?